src/target/target.c

```diff
@@ -770,6 +770,12 @@ static int jtag_enable_callback(enum jtag_event event, void *priv)
 	return target_examine_one(target);
 }
 
+/* When this is true, it's OK to call examine() again in the hopes that this time
+ * it will work.  Earlier than that there is probably other initialization that
+ * needs to happen (like scanning the JTAG chain) before examine should be
+ * called. */
+static bool examine_attempted;
```

Comment:
Out of curiosity, is there any plausible scenario where examine_attempted might need to be tracked on a per-target basis (e.g. as a data member of "struct target")?

Reply:
I don't believe so. Really I just want to make sure we don't call examine() from poll() until after the TAP scan chain has been examined. Hopefully the scan chain will always be a single thing, and examine() won't be called outside of poll() until after all chains (I don't think OpenOCD supports multiple chains even) have been examined. There isn't a clear place where that ordering happens that I could find, so I'm using this work-around.